### PR TITLE
Feat 32/statistics hj

### PR DESCRIPTION
**Pull request by sueelly**
## #️⃣ Related Issue

Linked Feature backlog: 

## 📝 Purpose of PR

Please briefly explain your work in this PR.

## Contents

List of work items performed.

## Checklist

-   [ ] Does commit messages follow the convention?
-   [ ] Are variable names brief but descriptive?
-   [ ] Is the code clean and easy to understand?
-   [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
-   [ ] Does the test successfully test edge and corner cases?

## 💬 Review Requirements (Optional)

Is there anything you can praise about this PR? Start with praise.
Make a review, leaving kind comments and suggesting changes where needed (to resolve the above).

### CODE REVIEW

File: src/main/java/ac/dnd/dodal/domain/goal/event/GoalCreatedEvent.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+@Getter
+@EqualsAndHashCode
+public class GoalCreatedEvent {
+
+    private Long goalId;
+
+    public GoalCreatedEvent(Long goalId) {
+        if (goalId == null) {
+            throw new IllegalArgumentException("GoalCreatedEvent: goalId is null");
```
Comment (5hseok) [inline src/main/java/ac/dnd/dodal/domain/goal/event/GoalCreatedEvent.java:14]: 여기서는 따로 GoalException은 만들지 않나요?

File: src/main/java/ac/dnd/dodal/domain/goal/model/GoalStatistics.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.EqualsAndHashCode;
+import ac.dnd.dodal.domain.plan.enums.PlanStatus;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+import jakarta.persistence.Id;
+
+@Entity(name = "goal_statistics")
```
Comment (5hseok) [inline src/main/java/ac/dnd/dodal/domain/goal/model/GoalStatistics.java:11]: ERD Cloud에서도 업데이트 부탁드릴게요!

File: src/main/java/ac/dnd/dodal/application/goal/service/GoalStatisticsEventListener.java
Comment (5hseok) [inline src/main/java/ac/dnd/dodal/application/goal/service/GoalStatisticsEventListener.java:1]: 오 이벤트 처리로 구현하셨군요!
어떤 문제상황을 해결하기 위해 도입하게 되었고, 어떻게 떠올리게 되었는 지가 궁금해요!

**Comment by 5hseok** [comment]
API 구현 상황에 따라 API 명세서에 아이콘 표기 부탁드릴게요~
배포 완료는 제가 바꿔놓을테니 작성할 명세 페이지에서 구현 완료, 구현 중 아이콘으로만 수정해주시면 될 것 같아요.

File: src/main/java/ac/dnd/dodal/domain/goal/event/GoalCreatedEvent.java (hunk 1 above)
Comment (sueelly) [reply src/main/java/ac/dnd/dodal/domain/goal/event/GoalCreatedEvent.java:14]: 기본적인 IllegalArgument는 따로 생성하지 않으려고요~

client에게 전달할 사항들을 위주로 만든다고 보시면 될 것 같아요

File: src/main/java/ac/dnd/dodal/domain/goal/model/GoalStatistics.java (hunk 2 above)
Comment (sueelly) [reply src/main/java/ac/dnd/dodal/domain/goal/model/GoalStatistics.java:11]: 오 안그래도 수정사항이 많아서 완료 후 한번에 업데이트하도록 하겠습니다!

File: src/main/java/ac/dnd/dodal/application/goal/service/GoalStatisticsEventListener.java
Comment (sueelly) [reply src/main/java/ac/dnd/dodal/application/goal/service/GoalStatisticsEventListener.java:1]: statistic은 조회를 위해 저장하는 테이블이에요. 저장하지 않아도 매번 모든 테이블을 조회하여 계산하여 가져올 수 있지만, 그러면 성능이 떨어질 것 같아서 따로 저장하는건데, 그러면 plan/goal에 command가 수행될 때 언제나 함께 업데이트됩니다.
전에 cqrs를 논했던 게 결국 command와 query용 db를 나누어 command가 수행되었을 때 이벤트로 query용 db를 업데이트하는 것인데, 이것을 db 레벨이 아니라 어플리케이션 레벨에서 처리했다고 보시면 될 것 같아요.
